### DISCUSSION TITLE
What's the best way to schedule materialization of a set of assets with different partitions?

### DISCUSSION THREAD

**Discussion by craustin** [dagster-io/dagster#19522] [Q&A]
I have a set of assets like the below each w/ different partitions and I'd like to schedule a daily materialization. If I click "Materialize All" in the UI, Dagster does the right thing 🎉 , albeit it uses a backfill. (We even get a nice explanation of what's going to happen: "Dagster will materialize all partitions downstream of the selected partitions for the selected assets, using separate runs as needed.")

OTOH, if I define a *job* that targets these assets (code below), I receive "Multiple partitioned assets exist in assets job 'xyz'. Selected assets must have the same partitions definitions...".

Since Dagster's UI does the right thing already, it seems promising that there may be a good way to schedule these runs. I think I can solve this w/ sensors, but that'll be messy. I could also have a cron outside of Dagster that triggers a backfill, just as the UI does. Is there another way?

```python
from dagster import AllPartitionMapping
from dagster import asset
from dagster import AssetDep
from dagster import AssetExecutionContext
from dagster import define_asset_job
from dagster import StaticPartitionMapping
from dagster import StaticPartitionsDefinition


@asset(
    partitions_def=StaticPartitionsDefinition(partition_keys=["a", "b"]),
)
def asset_1(context: AssetExecutionContext):
    context.log.info(f"asset_1: {context.partition_key}")


@asset(
    partitions_def=StaticPartitionsDefinition(partition_keys=["c", "d"]),
    deps=[AssetDep(asset_1, partition_mapping=AllPartitionMapping())],
)
def asset_2(context: AssetExecutionContext):
    context.log.info(f"asset_2: {context.partition_key}")


@asset(
    partitions_def=StaticPartitionsDefinition(partition_keys=["e", "f"]),
    deps=[AssetDep(asset_1, partition_mapping=StaticPartitionMapping({"a": "e", "b": "f"}))],
)
def asset_3(context: AssetExecutionContext):
    context.log.info(f"asset_3: {context.partition_key}")

# this job definition raises an exception
job = define_asset_job("repro_job", "asset_1*")
```

Related: https://github.com/dagster-io/dagster/issues/12580, https://github.com/dagster-io/dagster/issues/13257, https://github.com/dagster-io/dagster/discussions/15247, https://github.com/dagster-io/dagster/discussions/15675

**Comment by the4thamigo-uk** [comment]
Also interested in this https://github.com/dagster-io/dagster/discussions/19365
